modules/rpc/yarpc.go

```diff
@@ -183,3 +191,18 @@ func (m *YarpcModule) IsRunning() bool {
 	defer m.stateMu.RUnlock()
 	return m.rpc != nil
 }
+
+type yarpcDispatcherFn func(yarpc.Config) (yarpc.Dispatcher, error)
+
+// RegisterDispatcher allows you to override the YARPC dispatcher registration
+func RegisterDispatcher(dispatchFn yarpcDispatcherFn) {
+	_dispatcherFn = dispatchFn
```

Comment:
Panic on set when it's already non-nil, then? Just so we don't accidentally do it twice, or have default override the custom provided one.